logstash-core/lib/logstash/pipelines_registry.rb

```diff
@@ -189,11 +258,5 @@ def select_pipelines(&optional_state_filter)
         end
       end
     end
-
-    def get_lock(pipeline_id)
-      @locks.compute_if_absent(pipeline_id) do |k|
-        java.util.concurrent.locks.ReentrantLock.new
```

Comment:
We are replacing a reentrant lock with a regular mutex. Is there any situation where a thread could attempt to acquire a lock that it already holds?

Reply:
Good catch. No spec has failed from this removal so it is unclear if this is in fact needed. But for the safety purposes I will assume there is/was a good reason for that and change the `Mutex` for a `Monitor` which **is** reentrant to make sure we preserve the same semantic here. I will also add a comment + followup issue to try to get to the bottom of that.

Reply:
After verification, I do not see any case where a reentrant lock is required. It would be useful only if any of the `create_pipeline`, `reload_pipeline` or `terminate_pipeline` methods could   call into each other which is not the case. IIRC at some point the actions were calling into themselves, for example the reload action was calling the stop and create actions, but this is not the case anymore.